### PR TITLE
refactor: optional dependency resturcture in native-package

### PR DESCRIPTION
**Pull request by khushal87**
## 🎯 Goal

This PR deals with restructuring the optional dependency flow for native CLI in react native.

## 🛠 Implementation details

- "optional": true is added.
- Directory structure is improved

## 🎨 UI Changes

None

## 🧪 Testing

Not required

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Documentation is updated
- [x] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [x] Expo iOS and Android

### CODE REVIEW

**Review by vanGalilea — COMMENTED** [review]
Code delta LGTM, I just need to test through physical devices, to be continued

File: examples/SampleApp/ios/Podfile.lock [hunk 1]
```diff
@@ -860,7 +859,10 @@ SPEC CHECKSUMS:
   FlipperKit: cbdee19bdd4e7f05472a66ce290f1b729ba3cb86
   fmt: ff9d55029c625d3757ed641535fd4a75fedc7ce9
   glog: 04b94705f318337d7ead9e6d17c019bd9b1f6b1b
+<<<<<<< HEAD
```
Comment (vishalnarkhede) [inline examples/SampleApp/ios/Podfile.lock:862]: image

Comment (khushal87) [reply examples/SampleApp/ios/Podfile.lock:862]: Damn, let me fix it quickly.

File: package/native-package/package.json [hunk 2]
```diff
@@ -18,17 +18,22 @@
     "@react-native-community/netinfo": ">=2.0.7",
     "@stream-io/flat-list-mvcp": "^0.10.2",
     "react-native": ">=0.60.0",
-    "react-native-document-picker": ">=3.2.0",
     "react-native-fs": ">=2.16.6",
-    "react-native-haptic-feedback": ">=1.11.0",
     "react-native-image-crop-picker": ">=0.33.2",
-    "react-native-image-resizer": ">=1.4.2",
-    "react-native-share": ">=4.1.0",
-    "@react-native-clipboard/clipboard": "^1.11.1"
```
Comment (santhoshvai) [inline package/native-package/package.json:27]: it should not be removed from peer deps ⚠️

Comment (khushal87) [reply package/native-package/package.json]: Fixed @santhoshvai 😄

**Pull request merged**